### PR TITLE
Simulator api update

### LINKED ISSUE #4: Move send methods to simulationManager without circular dependency
https://github.com/renew-js/renew-simulator-ccpn/blob/0c6fdff2c0a2d9627ec9f53685ee6c510324d1b3/src/CCPNSimulatorPlugin.js#L98-L103

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in 0c6fdff2c0a2d9627ec9f53685ee6c510324d1b3 when #3 was merged. cc @renew-js.

### LINKED ISSUE #7: Stop requesting net markings when simulation has stopped
The newest version of ccpn introduces a boolean attribute `stopped` in the returned `netMarking`.
This information should be forwarded to the client.

### PR DESCRIPTION
**Pull request by 0x4d**
Closes #7 
Closes #4 

## Changes
- Use the new abstract plugin
- Send feedback when simulator is initialized
- Send feedback when Simulator has halted
- Add example with correct PNML positioning

### CODE REVIEW

File: src/CCPNSimulatorPlugin.js [hunk 1]
```diff
@@ -1,13 +1,25 @@
 const xmlBuilder = require('xmlbuilder');
 const xmlParser = require('fast-xml-parser');
+const AbstractPlugin =
+    require('renew-simulator-gateway/src/plugin/AbstractPlugin');
```
Comment (tim-kilian) [inline src/CCPNSimulatorPlugin.js:4]: Why are you using the require statement instead fo the import? I would prefer the import.

File: src/CCPNSimulatorPlugin.js [hunk 2]
```diff
@@ -17,45 +29,51 @@ class CCPNSimulatorPlugin {
...
+                        this.sendInitialized(socket);
+                    }
+                    this.updateMarking(socket, data[elementName]);
+                    break;
+            }
         });
     }
 
     initSimulation (formalismId, netInstance, serializedData) {
+        console.log(this.name + ': Initializing simulation ...');
```
Comment (tim-kilian) [inline src/CCPNSimulatorPlugin.js:68]: Remember this log

File: src/CCPNSimulatorPlugin.js [hunk 3]
```diff
@@ -71,37 +89,42 @@ class CCPNSimulatorPlugin {
...
+                text: '',
                 type: 'pt:marking',
+            };
+            if (placeState.token && placeState.token.text) {
+                element.text = placeState.token.text;
             }
+            return element;
         });
+        const isHalted = netMarking.attributes
+            && netMarking.attributes.stopped === 'True';
```
Comment (tim-kilian) [inline src/CCPNSimulatorPlugin.js:120]: 'True'...

**Review by tim-kilian — APPROVED** [review]

File: src/CCPNSimulatorPlugin.js (hunk 1 above)
Comment (0x4d) [reply src/CCPNSimulatorPlugin.js:4]: That would require using the `--experimental-modules` or a transpiler  wouldn't it?

File: src/CCPNSimulatorPlugin.js (hunk 3 above)
Comment (0x4d) [reply src/CCPNSimulatorPlugin.js:120]: Could not get fast-xml-parser to parse this as boolean :disappointed:

File: src/CCPNSimulatorPlugin.js (hunk 1 above)
Comment (tim-kilian) [reply src/CCPNSimulatorPlugin.js:4]: whats the case in not using a transpiler?

**Pull request merged**